src/cudadev/DataFormats/SoAStore.h

```diff
@@ -0,0 +1,569 @@
+                                                                                                                                    \
+  /**                                                                                                                               \
+   * Helper/friend class allowing SoA introspection.                                                                                \
+   */                                                                                                                               \
+  struct SoAMetadata {                                                                                                              \
+    friend CLASS;                                                                                                                   \
+    SOA_HOST_DEVICE_INLINE size_t size() const { return parent_.nElements_; }                                                       \
+    SOA_HOST_DEVICE_INLINE size_t byteSize() const { return parent_.byteSize_; }                                                    \
+    SOA_HOST_DEVICE_INLINE size_t byteAlignment() const { return parent_.byteAlignment_; }                                          \
+    SOA_HOST_DEVICE_INLINE std::byte* data() const { return parent_.mem_; }                                                         \
+    SOA_HOST_DEVICE_INLINE std::byte* nextByte() const { return parent_.mem_ + parent_.byteSize_; }                                 \
+    SOA_HOST_DEVICE_INLINE CLASS cloneToNewAddress(std::byte* addr) {                                                               \
+      return CLASS(addr, parent_.nElements_, parent_.byteAlignment_ );                                                              \
+    }                                                                                                                               \
+    _ITERATE_ON_ALL(_DEFINE_METADATA_MEMBERS, ~, __VA_ARGS__)                                                                       \
+                                                                                                                                    \
+  private:                                                                                                                          \
+    SOA_HOST_DEVICE_INLINE SoAMetadata(const CLASS& parent): parent_(parent) {}                                                     \
+    const CLASS& parent_;                                                                                                           \
+  };                                                                                                                                \
+  friend SoAMetadata;                                                                                                               \
+  SOA_HOST_DEVICE_INLINE const SoAMetadata soaMetadata() const { return  SoAMetadata(*this); }                                      \
```

Comment:
Confusing `const`
```suggestion
  SOA_HOST_DEVICE_INLINE SoAMetadata soaMetadata() const { return  SoAMetadata(*this); }                                      \
```

Reply:
`SoAMetadata` gives access to column addresses through the `addressOf_x()` member functions. Those are present in both views and layouts and are used internally to initialize view column addresses. The const version of a SoAView only delivers pointers to const data. This is still simple const correctness that can be bypassed by copy, but only applies to layouts and mutable views, for which const can already be bypassed by copy.

I completed the previous incomplete const correctness in the this area in the followup version.

The const views, on the opposite, are fully safe as they do not rely on const correctness.